### ISSUE TITLE
Ease of server making

### ISSUE THREAD

**Issue by xuuyz** [PrismLauncher/PrismLauncher#864]
### Role

I make servers

### Suggestion

Ability to quickly and easily make servers

### Benefit

I make servers for my friends and its very inconvenient making a server.

### This suggestion is unique

- [x] I have searched the issue tracker and did not find an issue describing my suggestion, especially not one that has been rejected.

### You may use the editor below to elaborate further.

I have made some mockups. 2 ideas.
1: Servers are a separate tab
2: Launch an instance as a server
![Screenshot_4](https://user-images.githubusercontent.com/63068548/218895671-36ece46e-4a14-4057-82ed-3d6a5fe77c94.png)
![Screenshot_3](https://user-images.githubusercontent.com/63068548/218895672-0a30d6ae-4913-47fe-9f4d-b8c3ff29f2a2.png)
![e](https://user-images.githubusercontent.com/63068548/218895673-6a6b41cb-7a12-4dd9-98d1-db4c176d8fd3.png)

**Comment by ghost** [comment]
It would feel good to have server instances. They could just be marked with an indicator near their icons, a whole new tab would be a bit too overkill.

When creating an instance, there could be a "Server" checkbox somewhere in a corner, maybe the bottom left corner, under the platform list? It would be ideal to also have a few additional server softwares listed there if that's checked, that is, paper, purpur, all that...

**Comment by Xetaxheb** [comment]
Servers would need to be separate instances entirely, using the server pack files authors provide. There are clientside mods that will crash servers if they are in /mods/ and prism has no reasonable way to identify these. Adding support to download server packs would be nice, however (cf) server packs are not _modpacks_, they are redistributions of all jar and other necessary files and configs.

Other concerns:
 * Cf reviews the zips manually so the batch files shouldn't be any more of a security risk than usual modding, right?
 * Jar redistribution sits in a grey area curseforge kind of ignores the implications of for now afaik? I'm not sure it's worth getting into it with cf if attention is drawn to these because of it.
 * Prism isn't really a server management program and even if the fabric/forge server installers were run there's not that much utility to be gotten out of doing this through prism instead of the easy jars server packs provide.
 * Running a client install as a server can have a host of problems **especially** if it's concurrently.

**Comment by ghost** [comment]
CF server packs are screwed and why'd we care that much

We're making a power user launcher here so if people are stupid it's their business, if you're going into server creation you should know how to deal with servers anyway. Servers aren't much different from clients, especially when it comes to mod loaders.

**Comment by P3rf3ctXZer0** [comment]
I think its the pack builders responsibility to test if a mod is server client or both. Is very unrealistic to automate this. This launcher should install the necessary base launcher files and libraries but a popup should still show Eula for legal protection of Prism. You should have a global black and whitelist for known client and server mods but it should be entirely optional. It should be easy for a person to set a mod as blacklist even as wildcard when doing pack creation and it would be nice if the not functionality could be on a per pack per version and maybe even per mod basis. This would allow faster trouble shooting. Example I know canary breaks Minecolonies so I could add this category to notes canary*.jar [This mod is incompatible with Minecolonies on a fundamental level.] Users should be allowed to commit this knowledge so we can pool our information together. Also the launch button should run whatever start apparatus the user sets examples .bat .exe .hamster you get the idea. It should need to be configured on a per pack per user basis.

**Comment by robinofski-dev** [comment]
I want to try here before opening a new issue, since I think it relates to "Ease of server making". How about we show in the table of mods in a profile, if they're server-,  client-side or both? It would fit behind the version number column and would help eliminate a lot of hassle, looking for the mods I need to delete.
I found that the internal mod menu that minecraft has when launching it with fabric, has client labels attached to mods. But this was not correct all the time. The logs also aren't much help. And, because of the resetting pagination after showing mod details, it's quite a hassle to check every mod for that clientside / serverside indication on the detail page.

**Comment by snkiz** [comment]
I'm doing this. As a 'power user' I am taking on the responsibility of vetting that mods I'm downloading are compatible with servers.
- First I created an instance in Prism with the same fabric version as in my server instance. (I used simply optimized as I had it on the server.)
- Then I Used Flatseal to give prism rw access to the server instances folder. 
- Then I deleted the mod and config folders in the prism instance and replaced them with symlinks to the server equivalents.
- I restarted prism, and updated the mods. Prism regenerated the index file. Everything worked.

I was pleasantly surprised to find prism recognized that these folders are now symlinks, though I miss the mod icons. This gives me the added bonus of being able to run a local client on prism for testing configurations. Since prism con only see those two folders nothing else is affected and prism still has it's own directory. 
A really simple solution would be to add a button that could automate this on prism's end. Flatseal would still have to be done manually since prism doesn't know where a users server install is by default. A tooltip can solve that.

**Comment by P3rf3ctXZer0** [comment]
> I'm doing this. As a 'power user' I am taking on the responsibility of vetting that mods I'm downloading are compatible with servers.
> 
> * First I created an instance in Prism with the same fabric version as in my server instance. (I used simply optimized as I had it on the server.)
> * Then I Used Flatseal to give prism rw access to the server instances folder.
> * Then I deleted the mod and config folders in the prism instance and replaced them with symlinks to the server equivalents.
> * I restarted prism, and updated the mods. Prism regenerated the index file. Everything worked.
> 
> I was pleasantly surprised to find prism recognized that these folders are now symlinks, though I miss the mod icons. This gives me the added bonus of being able to run a local client on prism for testing configurations. Since prism con only see those two folders nothing else is affected and prism still has it's own directory. A really simple solution would be to add a button that could automate this on prism's end. Flatseal would still have to be done manually since prism doesn't know where a users server install is by default. A tooltip can solve that.

I have just doing a work around with ATLauncher Prism Launcher Combo at this point on a vm.

**Comment by kadincool** [comment]
+1 on this
it would be nice if there was an option to duplicate an instance as a server instance, filter all modrinth mods (and curseforge if/when they add the metadata) and remove the client side only mods from an instance to help make servers of modpacks.

**Comment by aflyingpumpkin** [comment]
Yo! If anyone else is here trying to make a server I made a little bash utility to automate the process of finding mods that don't work on servers.
```sh
#!/bin/sh

# To get modlist.csv go to yourpack > edit > Mods > Export modslist, and set the string to {url},{name},{filename}

while IFS=, read -r url name file; do
    id=${url: -8}
    json="$(curl -s "https://api.modrinth.com/v2/project/$id")"
    if [[ $(echo "$json" | jq ."server_side" | grep unsupported) ]]; then
        echo "$name is unsupported, file: $file"
        # rm mods/$file ## Uncomment this line to actually remove the mods instead of just saying which ones do it.
    fi
done < modlist.csv
```

**worth noting I used the api endpoint for individual projects in a loop instead of writing better code to bulk get all mods from the api. Also didn't write any handling if some mods are from curseforge since none of mine are.**

**Comment by qtqgyt** [comment]
I don't believe this would be ever added, but I am adding my vote.

**Comment by Spock117** [comment]
We need this!

**Comment by Davvv1** [comment]
This is something I would _really_ like to have.

**Comment by Xetaxheb** [comment]
This isn't a democracy, there is no voting, no more "me too" posts

If you want the feature, code it well and make a pull request

If you have (actionable, knowledge and code-based) solutions to the problems brought up, 

* Clientside mods that aren't meta tagged as such and crash the server
* Mods improperly meta or distributor tagged that work fine on server
* Server-pack only mods that are required to run a standalone server (looking at you polymer)
* Poorly thought out modpacks that are not singleplayer compatible because they're designed to run playing on a server with server-pack only data (hello create astral[? or one of similarly named; maybe C:AAB] and its ftbquest serverside updates)
* Datapacks that have nowhere to go until world is created
* TOS gray area of cf server file distribution
* NF/F/F/Q server jar distribution and installation
* Functionality in prism to handle a server (hello, no console input??)
* Concurrent file access (aka, a server needs separate files from the client)
* Filesystem parity with launching outside of prism (that is, you cannot change how directories point or are 'symlinked' or interpreted, one needs to be able to copy paste the minecraft folder to say overwolf or a standalone install and have it just work; so no separating /config/ into /config/client/ and /config/server/ let alone hardcoded subdirectories in mods)
* Whatever integrations with paper/spigot based servers and plugins
* Making it meaningfully more useful than just copy-pasting /mods/, /config/, and other relevant folders to an install

then great, otherwise stop spamming

**Comment by Davvv1** [comment]
> If you have (actionable, knowledge and code-based) solutions to the problems brought up, 
> 
> * Clientside mods that aren't meta tagged as such and crash the server
> * Mods improperly meta or distributor tagged that work fine on server
> * Server-pack only mods that are required to run a standalone server (looking at you polymer)
> * Poorly thought out modpacks that are not singleplayer compatible because they're designed to run playing on a server with server-pack only data (hello create astral[? or one of similarly named; maybe C:AAB] and its ftbquest serverside updates)
> * Concurrent file access (aka, a server needs separate files from the client)
> * Filesystem parity with launching outside of prism (that is, you cannot change how directories point or are 'symlinked' or interpreted, one needs to be able to copy paste the minecraft folder to say overwolf or a standalone install and have it just work; so no separating /config/ into /config/client/ and /config/server/ let alone hardcoded subdirectories in mods)
> * Making it meaningfully more useful than just copy-pasting /mods/, /config/, and other relevant folders to an install

A lot of the issues you mentioned (the ones I left in the quote) seem to have a single solution as simple as a "Make Into Server" button on the instance, with it making another server "instance" after choosing which mods you want to be copied over. It could automatically disable mods that have singleplayer only or client-side tag with no server tag, and you could enable them after you make the server. Of course there would still be a lot of trial and error, but much less than there would otherwise be.

I don't have that much experience with actual development, but I have enough general information to know that this could probably work. I hope someone more knowledgeable than me will make this.

**Comment by Davvv1** [comment]
You could also make the user download the server zip/jar themselves, and prism could then use that to make the server instance, thus defeating the TOS* (I think, I should probably read it when I get home) and just make it a normal server that prism can launch.

**Comment by Xetaxheb** [comment]
> I don't have that much experience with actual development, but I have enough general information to know that this could probably work. I hope someone more knowledgeable than me will make this.

I do have more than plenty actual development experience, and as I said, if you are not contributing the code to this, **please stop spamming**
with all due respect, your reply does **not** contain literally anything useful and is just another "me too"
there is no use to any more "ideas" posts, the people who can create this already know what needs doing
it is not a complicated thing to do, there are issues that do not and will not have good (non-manual, ethically [tos] usable, valuable) resolutions because of modded distribution infrastructure and standards as a whole
realistically those who _fully_ understand the situation and can do something about it realize it is _not worth the effort_
so until someone comes along and wants to put in the elbow grease to make the only obvious solutions on this pet project, it will not happen

TL;DR this is not at all _complicated_ to implement any of this at a basic notion, _it is a ton of work_ (and will cause way too many user reported issues), _it is not worth it_ so nobody has

if anyone feels differently, feel free to submit a PR

**Comment by LZeugirdor** [comment]
+1 for this

**Comment by seasea1010** [comment]
That's the feature I wanted most.

**Comment by qtqgyt** [comment]
Again, while I assume that this PROBABLY won't be implemented anytime soon, I do want to give a +1 to this

**Comment by random817** [comment]
> This isn't a democracy, there is no voting, no more "me too" posts
> 
> If you want the feature, code it well and make a pull request
> 
> If you have (actionable, knowledge and code-based) solutions to the problems brought up,
> 
>     * Clientside mods that aren't meta tagged as such and crash the server
> 
>     * Mods improperly meta or distributor tagged that work fine on server
> 
>     * Server-pack only mods that are required to run a standalone server (looking at you polymer)
> 
>     * Poorly thought out modpacks that are not singleplayer compatible because they're designed to run playing on a server with server-pack only data (hello create astral[? or one of similarly named; maybe C:AAB] and its ftbquest serverside updates)
> 
>     * Datapacks that have nowhere to go until world is created
> 
>     * TOS gray area of cf server file distribution
> 
>     * NF/F/F/Q server jar distribution and installation
> 
>     * Functionality in prism to handle a server (hello, no console input??)
> 
>     * Concurrent file access (aka, a server needs separate files from the client)
> 
>     * Filesystem parity with launching outside of prism (that is, you cannot change how directories point or are 'symlinked' or interpreted, one needs to be able to copy paste the minecraft folder to say overwolf or a standalone install and have it just work; so no separating /config/ into /config/client/ and /config/server/ let alone hardcoded subdirectories in mods)
> 
>     * Whatever integrations with paper/spigot based servers and plugins
> 
>     * Making it meaningfully more useful than just copy-pasting /mods/, /config/, and other relevant folders to an install
> 
> 
> then great, otherwise stop spamming

GUYS PLEASE LEARN HOW TO READ FOR THE LOVE OF GOD

**Comment by qtqgyt** [comment]
> > This isn't a democracy, there is no voting, no more "me too" posts
> > If you want the feature, code it well and make a pull request
> > If you have (actionable, knowledge and code-based) solutions to the problems brought up,
> > ```
> > * Clientside mods that aren't meta tagged as such and crash the server
> > 
> > * Mods improperly meta or distributor tagged that work fine on server
> > 
> > * Server-pack only mods that are required to run a standalone server (looking at you polymer)
> > 
> > * Poorly thought out modpacks that are not singleplayer compatible because they're designed to run playing on a server with server-pack only data (hello create astral[? or one of similarly named; maybe C:AAB] and its ftbquest serverside updates)
> > 
> > * Datapacks that have nowhere to go until world is created
> > 
> > * TOS gray area of cf server file distribution
> > 
> > * NF/F/F/Q server jar distribution and installation
> > 
> > * Functionality in prism to handle a server (hello, no console input??)
> > 
> > * Concurrent file access (aka, a server needs separate files from the client)
> > 
> > * Filesystem parity with launching outside of prism (that is, you cannot change how directories point or are 'symlinked' or interpreted, one needs to be able to copy paste the minecraft folder to say overwolf or a standalone install and have it just work; so no separating /config/ into /config/client/ and /config/server/ let alone hardcoded subdirectories in mods)
> > 
> > * Whatever integrations with paper/spigot based servers and plugins
> > 
> > * Making it meaningfully more useful than just copy-pasting /mods/, /config/, and other relevant folders to an install
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > then great, otherwise stop spamming
> 
> GUYS PLEASE LEARN HOW TO READ FOR THE LOVE OF GOD

Apologies, I cannot read.
Regarding the TOS gray area, I propose doing something like Fabric and Paper do, instead of distributing the server jar ourselves, download the server jar officially, like Fabric and Paper do.
Sorry, that doesn't make any sense. I cannot type either.

Additionally, regarding clientside mods not marked so, atleast in my experience, I have never had a client side mod crash a server. It just doesn't do anything.

**Comment by Xetaxheb** [comment]
> Apologies, I cannot read. Regarding the TOS gray area, I propose doing something like Fabric and Paper do, instead of distributing the server jar ourselves, download the server jar officially, like Fabric and Paper do. Sorry, that doesn't make any sense. I cannot type either.
> 
> Additionally, regarding clientside mods not marked so, atleast in my experience, I have never had a client side mod crash a server. It just doesn't do anything.

I could have phrased it better but you've misunderstood: 
Distributing the server files a la neoforge/fabric server jars is not the problem here. Wouldn't make a difference to download it from source anyway since it's either in the server zip already or done through startscripts.
 
That's about the fact that curseforge server zips include every .jar file from **_every mod_** which is definitely a violation of copyright besides curseforge tos carveouts about them owning you. There's been legal issues in the past about uploaders including things that aren't first-party distributed on curseforge. But I significantly doubt curseforge will tolerate 3rd party server zip downloads if it starts happening a lot.

### At any rate, please stop spamming unless it's a link to a pull request with your submitted code. 
### There is NO USE to any +1 messages, they do nothing.
### If you're not going to implement it yourself or pay someone else fair market price for their work, then it's not going to happen.

**Comment by snkiz** [comment]
Forgive me, but this is tagged as needs discussion. Asking for cash or code is a bit much when a plan hasn't even been fleshed out, or anyone willing to spearhead one, apparently. IMO your list of challenges has a few to many guardrails and assumptions about the user's intent. I think the scope is a bit bigger then what what most people are asking for. 

So is the topic open for discussion or not? Sure it's not a democracy, but then why even invite comment if that's the attitude? If you aren't going to shut down the idea, maybe a better way to stop the me 2 posts would be a road-map with smaller goals individuals could achieve, a minimum viable project to build on.
I think out of frustration you've thrown the kitchen sink of 'what ifs' at this and that's no way to encourage new contributors.

**Comment by Xetaxheb** [comment]
> Forgive me, but this is tagged as needs discussion. Asking for cash or code is a bit much when a plan hasn't even been fleshed out, or anyone willing to spearhead one, apparently. IMO your list of challenges has a few to many guardrails and assumptions about the user's intent. I think the scope is a bit bigger then what what most people are asking for.
> 
> So is the topic open for discussion or not? Sure it's not a democracy, but then why even invite comment if that's the attitude? If you aren't going to shut down the idea, maybe a better way to stop the me 2 posts would be a road-map with smaller goals individuals could achieve, a minimum viable project to build on. I think out of frustration you've thrown the kitchen sink of 'what ifs' at this and that's no way to encourage new contributors.

Yes... discussion about useful info not spam. As I said:
> **If you have (actionable, knowledge and code-based) solutions to the problems brought up, then great, otherwise stop spamming**

You have not touched on solving a single point made, yet somehow recognize that nobody is willing or interested in proceeding with doing all the grindwork on this. So, yet again, please ***stop spamming***. If you think any of the issues brought don't have merit then ***explain why they ought not be considered problems and what the solution should look like without them***. Simply expressing that you don't understand _why_ they're problems is _not helping_.

I'd advise you to keep in mind the reality of the userbase when considering what you think is a problem or not. The influx of issue spam with people using a too-basic solution not understanding what's going on past "my server wont run" is also worth mentioning.

> * Clientside mods that aren't meta tagged as such and crash the server
> * Mods improperly meta or distributor tagged that work fine on server
> * Server-pack only mods that are required to run a standalone server (looking at you polymer)
> * Poorly thought out modpacks that are not singleplayer compatible because they're designed to run playing on a server with server-pack only data (hello create astral[? or one of similarly named; maybe C:AAB] and its ftbquest serverside updates)

These problems existing all pretty much immediately remove 99% of the utility and turn it into a glorified and _**opaque**_  copy/paste event that if anything harms the novice users understanding and makes it far less useful to the knowledgeable users. There are not clean solutions here because of garbage in garbage out; flawed jars are gonna be flawed and it takes manual review when creating a server to weed it out. If prism is not making that process easier, then it is a waste of time.

> * Datapacks that have nowhere to go until world is created
> * NF/F/F/Q server jar distribution and installation
> * Functionality in prism to handle a server (hello, no console input??)
> * Concurrent file access (aka, a server needs separate files from the client)
> * Filesystem parity with launching outside of prism (that is, you cannot change how directories point or are 'symlinked' or interpreted, one needs to be able to copy paste the minecraft folder to say overwolf or a standalone install and have it just work; so no separating /config/ into /config/client/ and /config/server/ let alone hardcoded subdirectories in mods)
> * Whatever integrations with paper/spigot based servers and plugins

These are the technical/implementation issues that need clean resolutions and a ton of busywork in implementing them. You're clearly either not motivated *or* not knowledgeable enough to do so, so who will? Unless you're going to contribute resources towards the progress in the form of time, knowledge, labor, financial incentives, or networking; why post? "Me too"/"I want this" does nothing, that sentiment has already been established plenty.

> * TOS gray area of cf server file distribution

This is definitely a huge issue with the entire concept, possibly the biggest: There is literally no incentives to invite smoke from cf for treading anywhere near this. The project ignoring cf's wishes about their API is already enough to cause massive headaches and the fact it's lazily tolerated should be sufficient. There's not even a point when server files ARE provided because they come self-contained, and when they're not provided all of the above issues still apply. I do admit that this problem of "downloading and running cf server packs" is not 100% the same as this issue, but there are so many bland/minimal issue posts that this is basically part of this undertaking at this point. Building out all the prism infrastructure to handle running servers without tackling this issue ("being able to download and run server packs from cf") or vice versa would be mostly pointless; nobody capable of running a server through prism shouldn't be able to just run it with the included run.bat/sh that server packs come with. It impedes new user knowledge growth, and serves no utility to knowledgeable ones.


> * Making it meaningfully more useful than just copy-pasting /mods/, /config/, and other relevant folders to an install

So again: **Unless you can demonstrate in clear, defined, and elaborated ways *why* and *how* some part serves a purpose (not "I want it" but rather "it would solve this problem if it could be used to take steps 1 2 3 instead of 4 5 6 7 8 9") then you are not *contributing* to the conversation.**



---
Even if, and this is a **_BIG IF_**, you stripped the entire concept down to a glorified "copy the directory and disable any client-only-tagged mods"..... why is that even useful? Someone with the knowledge to untangle the mess from there wouldn't need it, and someone without is just going to find it more difficult to learn because the process is opaque. 

Separately to this, but related enough to mention: the notion of prism creating some kind of masterlist for "mods we know don't run on a server"..... ok? Rimworld and skyrim and a few others maintained enormous compatibility (well, sorting) lists like this and it's ***a huge undertaking***.... who's going to do that busywork? How do you verify correctness of entries? Prevent biased/malicious entries? Keep up to date with mod updates that change the correctness? Etc etc etc. If you want to go make your own website/tool that does more research than a med student you are more than welcome...... or just go ask mod authors to correctly tag their projects as you come across the issue and solve it for everyone? Or convince cf/modrinth you know better than the developer and get the power to manually change tags. I just hope you're 100% accurate and not confused by mod conflicts doing a mixin into something that causes a gui class to fail to load in a separate mod making it look incompatible with servers when it normally would have silently proceeded and/or is required serverside because of channels etc etc etc.

---
## TL;DR:  yes... if you can provide ***useful discussion*** you are more than welcome to address any of the issues. Small talk about you wanting something is not a discussion. Things that can't be responded to or are not _**pertinent**_ or do not provide solutions are not discussion. 
## As @snkiz says: "...when a plan hasn't even been fleshed out..." is exactly correct...  _"me too" or "I want this" does not forward fleshing out the plan._

I'm pretty sure I already stripped it down to the bare minimum besides perhaps bukkit plugins. 
I don't believe the concept as a whole is really an addition that supports, endorses, nor is compatible with "small steps". 
One could argue separating the 
- "server running"
- "serverpack downloading"
- "client->server conversion" 

parts, but these are so intertwined in infrastructure/purpose/function and I see them as individually pointless 
- unnecessary overhead to get... log highlighting??
- ersatz web browser and 7z??
- worse alternative to copy/paste for newbies; enormous likelihood of errors both false positive and false negative

If you disagree and understand enough to, feel free to ***create said roadmap***.

**Comment by LZeugirdor** [comment]
> > Apologies, I cannot read. Regarding the TOS gray area, I propose doing something like Fabric and Paper do, instead of distributing the server jar ourselves, download the server jar officially, like Fabric and Paper do. Sorry, that doesn't make any sense. I cannot type either.
> > Additionally, regarding clientside mods not marked so, atleast in my experience, I have never had a client side mod crash a server. It just doesn't do anything.
> 
> I could have phrased it better but you've misunderstood: Distributing the server files a la neoforge/fabric server jars is not the problem here. Wouldn't make a difference to download it from source anyway since it's either in the server zip already or done through startscripts.
> 
> That's about the fact that curseforge server zips include every .jar file from **_every mod_** which is definitely a violation of copyright besides curseforge tos carveouts about them owning you. There's been legal issues in the past about uploaders including things that aren't first-party distributed on curseforge. But I significantly doubt curseforge will tolerate 3rd party server zip downloads if it starts happening a lot.
> ### At any rate, please stop spamming unless it's a link to a pull request with your submitted code.
> ### There is NO USE to any +1 messages, they do nothing.
> ### If you're not going to implement it yourself or pay someone else fair market price for their work, then it's not going to happen.

So, I'm broke, I can't support developers in the way I would like, that doesn't make my opinion any more or less valuable than anyone else's though. That said, +1 messages are useful, as they give a visible, measurable way to see the demand for the feature people want added. It can help developers gauge whether a feature is worth implementing or not, as in most cases, people are motivated by what helps the most people. This feature is very valuable and has been implemented a number of times in other projects, so it's not impossible, though may pose interesting challenges when it comes to parsing server and client mods. It really comes down to how much information platforms like curseforge give the application, and whether or not the application can understand that information.

Also some people use issues for suggestions, as well as issues, and give templates for both... which this repository does. So you're wrong here, +1 messages are the intended function when it comes to suggestions like this.

**Comment by Xetaxheb** [comment]
> So, I'm broke, I can't support developers in the way I would like, that doesn't make my opinion any more or less valuable than anyone else's though. That said, +1 messages are useful, as they give a visible, measurable way to see the demand for the feature people want added. It can help developers gauge whether a feature is worth implementing or not, as in most cases, people are motivated by what helps the most people. This feature is very valuable and has been implemented a number of times in other projects, so it's not impossible, though may pose interesting challenges when it comes to parsing server and client mods. It really comes down to how much information platforms like curseforge give the application, and whether or not the application can understand that information.
> 
> Also some people use issues for suggestions, as well as issues, and give templates for both... which this repository does. So you're wrong here, +1 messages are the intended function when it comes to suggestions like this.

So broke you can't afford to pay attention? Just ignoring the other 95% of the post that has nothing to do with contributing financially? 
**_+1 messages are useless, demand has nothing to do with the problem._** The feature is not worth implementing for reasons covered in all the text you ignored, _regardless of demand_. 
If the feature is implemented in so many other projects then either _go use those projects_ and stop spamming or _provide examples_ instead of spamming? I guarantee you it (holistically and accurately converting modded mc client list into server pack) hasn't been implemented in any other project...  you're speaking nonsense. If I'm wrong feel free to enlighten me. With links.

**_And oh look, you brought up exactly why it's a problem and has been said repeatedly:_** 
> It really comes down to how much information platforms like curseforge give the application, and whether or not the application can understand that information.

because curseforge DOESN"T GIVE THAT INFO and NEITHER DOES MODRINTH because NEITHER OF THEM ARE RESPONSIBLE FOR THAT, it's on MOD AUTHORS who DON'T DO IT PROPERLY in enough of a % cases that the prism client will just make it more arduous to convert. Do those platforms already provide the tools for mod authors to do so? YES. Do mod authors? NOT RELIABLY ENOUGH.
The day modrinth/cf run client/server unittests (in isolation with depts) to verify a side boots as described before marking an upload verified is the day raptor jesus comes to ascend us all.

If you want to +1... USE THE REACTION FEATURE ???
Your opinion isn't less valuable because you can't financially contribute... it's less valuable because you didn't think or read and digest even a fraction of the existing issue and have either no understanding of the problems involved or decide to just not think critically at all before posting. (But more accurately: NO opinions matter, we need solutions not opinions.)

## Stop spamming! 
You didn't contribute anything to the discussion and just said that you're here to share your empty opinion that has nothing to do with the problem??? Do you see why it's frustrating when people spam like that??? At least some of the responses that misunderstand the issues involved attempt to proffer solutions, even if they're barking up the wrong tree. But your "opinion" is not useful here at all. We don't need "opinion"s, we need code or novel solutions.

## Please do not post unless you have a solution to the existing problems or can enunciate and clearly lay out why you believe they are not problems and what the result would look like without those features/impediments.
